Hide window for typing when using i3wm.

tuxlifan opened the pull request:
Tested with i3wm and icewm on Linux.

tuxlifan committed:
Hide window for typing when using i3wm.

Tested with i3wm and icewm on Linux.

nthdimtech commented:
Is there a reason why doing minimization only is a problem on i3wm? I know some users prefer to have a task bar item open as it's easier to click and find than the task tray icon. You can always foreground with the device button but I could imagine not all users knowing or discovering that.

tuxlifan commented:
> Is there a reason why doing minimization only is a problem on i3wm?

When calling showMinimized() with i3wm the window doesn't change nor does it give away the focus.

The main cause I believe is that i3wm is a tiling window manager, so windows are supposed to be displayed somewhere on screen (according to the current tiled layout) or hidden.
There is no task bar with individual window placeholders as with other WMs but you can have a system tray area included in an information bar.

> it's easier to click and find [a task bar item] than the task tray icon. 

Indeed, especially on Windows where almost every program will have its own tray icon...

While the push-button-to-activate-signet-window feature is wonderful (and therefore it didn't even occur to me that someone might _not_ be using it) I see now that hiding the window and removing it thus from the task bar can be a surprise and/or inconvenience for many users.

The solution could be to either somehow try to detect if we are running under a tiling window manager (which is probably very error prone, if possible to do at all in a portable way) or to add a setting.

I will implement a new `Minimize window to system tray` setting.

For completeness of discussion:
1. I think the previous change in `SignetApplication::trayActivated()` can stay as it is since a click on the tray icon usually fully hides or restores the application window in my experience.
2. I'll patch the recently added `Minimize window` in the `File` menu to also use `background()` instead of `hide()` and thus honor the new setting.

tuxlifan committed:
Add "Minimize window to system tray" setting.

If set it will call hide() from background() to fully hide the window
(which will make it also disappear from task bar),
e.g. for use with tiling window managers.

setting name: minimizeToTray

tuxlifan committed:
Use background() i.s.o. hide() for "Minimize window" action in File menu

This honors the new "Minimize window to system tray" setting.

tuxlifan committed:
Add key accelerators for main functionalities in Settings

tuxlifan committed:
Add key accelerators for buttons when (re-)configuring the keyboard layout

tuxlifan committed:
Fix whitespace
